src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -83,15 +233,134 @@ export const ChannelView: React.FC = () => {
             )}
           </TitleContainer>
           <StyledButtonContainer>
-            <Button variant={isFollowing ? 'secondary' : 'primary'} onClick={handleFollow} size="large">
+            <StyledButton
+              icon={isFollowing ? <SvgGlyphCheck /> : <SvgGlyphPlus />}
+              variant={isFollowing ? 'secondary' : 'primary'}
+              onClick={handleFollow}
+              size="large"
+            >
               {isFollowing ? 'Unfollow' : 'Follow'}
-            </Button>
+            </StyledButton>
           </StyledButtonContainer>
         </TitleSection>
-      </Header>
-      <VideoSection className={transitions.names.slide}>
-        <InfiniteVideoGrid channelId={id} showChannel={false} />
-      </VideoSection>
+        <TabsContainer>
+          <StyledTabs
+            selected={isSearching ? -1 : TABS.findIndex((x) => x === currentTabName)}
+            initialIndex={0}
+            tabs={mappedTabs}
+            onSelectTab={handleSetCurrentTab}
+          />
+          {currentTabName === 'Videos' && (
+            <SearchContainer>
+              <StyledTextField
+                ref={searchInputRef}
+                isOpen={isSearchInputOpen}
+                value={searchQuery}
+                onChange={(e) => setSearchQuery(e.target.value)}
+                onKeyDown={handleSearchInputKeyPress}
+                placeholder="Search"
+                type="search"
+                helperText={null}
+              />
```

Comment:
We need to better separate concerns in this component. Because this is all a single component and the data flows the way it does, every keystroke inside the search field will cause the entire `ChannelView` to re-render. Try holding a single key inside the input and see what happens - the page becomes unresponsive. If you change the data flow and keep state of this input inside a separate component and only pass value to `ChannelView` on submit, we can avoid a lot of unnecessary re-rendering